### PR TITLE
Add Rebrandly shorten integration

### PR DESCRIPTION
**Pull request by StormFireFox1**
Adds "!shorten" command of the following format:

`!shorten <short tag> <long url> [description]`

Requires config change with two new values.

### CODE REVIEW

File: modules/shorten.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        // Options for request
+        const addLinkOptions = {
+          uri: 'https://api.rebrandly.com/v1/links',
+          method: 'POST',
+          headers: {
+            'apikey': config.rebrandlyAPIKey,
+            'Content-Type': 'application/json'
+          },
+          body: {
+            title: args[2] || 'Discord Bot - ' + args[0], // optional argument or slashtag
```
Comment (sumeet-bansal) [inline modules/shorten.js:66]: nit: assign the `args` elements to variables so these expressions are clearer, e.g.
```javascript
const [shorturl, longurl] = args;

const shorturl = args[0];
const longurl = args[1];
```

File: modules/shorten.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+          simple: false // we need this for 403 status code in request
+        };
+
+        rp(addLinkOptions)
+          .then(function (response) {
+            if (response.statusCode == 403) { // short link already exists
+              message.channel.send("A short link with the same name already exists! Try another one.");
+            } else {
+              // This makes a nice embed with a clickable URL for quick testing.
+              let shortenEmbed = new Discord.RichEmbed()
```
Comment (sumeet-bansal) [inline modules/shorten.js:84]: nice

File: modules/shorten.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                .setTitle("Set shortened link!")
+                .setDescription("Short link: " + response.body.shortUrl)
+                .setURL("https://" + response.body.shortUrl)
+                .setColor('0x3498DB');
+              message.channel.send(shortenEmbed);
+            }
+          })
+          .catch(function (err) {
+            message.channel.send("Error processing API request to Rebrandly! Check console logs.");
```
Comment (sumeet-bansal) [inline modules/shorten.js:93]: Most users aren't gonna check console logs; use more accessible error messages (or maybe @ yourself to bring any failed requests to your attention).

File: server.js [hunk 4]
```diff
@@ -13,12 +13,22 @@ global.Discord = require('discord.js');
...
 const express = require('express');
 const app = express();
+require('heroku-self-ping').default(config.herokuUrl);
 
 const port = process.env.PORT || 3000;
+
+// HTTP request for pinging
+//
+// Used to circumvent Heroku idling policy
+app.get('/ping', function(req, res){
```
Comment (sumeet-bansal) [inline server.js:23]: Free Heroku dynos have an uptime restriction (see [here](https://devcenter.heroku.com/articles/free-dyno-hours)), so if you keep pinging it to keep it up, you're gonna run out of those hours quickly and take down the other apps we have running on Heroku as well. Either we pay for the Heroku dyno that runs the bot or we look elsewhere for hosting.

File: modules/shorten.js (hunk 3 above)
Comment (StormFireFox1) [reply modules/shorten.js:93]: I think an error reporting platform might work in our favor here in the future (i.e. Sentry) to track these kinds of errors. I don't think dumping 500 server errors in the Discord server is a good idea.

For now, I'll just dump err.cause in a code block so it's formatted nicely and probably @ anyone in the dev team (I'll probably refrain from @'ing, though).

File: server.js (hunk 4 above)
Comment (StormFireFox1) [reply server.js:23]: AWS or GCP, I suppose, but if not, I suppose we'll just hit the "/ping" endpoint at 4:29 AM using a cronjob in order to maintain the channel wipe functionality.

File: modules/shorten.js (hunk 1 above)
Comment (StormFireFox1) [reply modules/shorten.js]: I've updated this part, you can check 0ea150e.

File: server.js (hunk 4 above)
Comment (StormFireFox1) [reply server.js:22]: I've removed the self-ping functionality for now, check c661598.

File: modules/shorten.js (hunk 3 above)
Comment (StormFireFox1) [reply modules/shorten.js]: I've added some more exposed error handling, see b0e1d42.

File: modules/shorten.js (hunk 1 above)
Comment (StormFireFox1) [reply modules/shorten.js]: I've made expressions clearer, see 0ea150e.

**Comment by StormFireFox1** [comment]
I've more or less addressed most of the issues. I'll handle the linting in a different PR.

File: server.js [hunk 5]
```diff
@@ -15,13 +15,22 @@ const express = require('express');
...
 
 const port = process.env.PORT || 3000;
+
+// HTTP request for pinging
+//
+// Potentially used for keeping bot alive.
+app.get('/ping', function(req, res){
+   res.send("pong");
+});
+
```
Comment (sumeet-bansal) [inline server.js:25]: delete unused route

File: package.json [hunk 6]
```diff
@@ -20,6 +20,7 @@
   "dependencies": {
     "discord.js": "^11.5.1",
     "express": "^4.17.1",
+    "heroku-self-ping": "^2.0.1",
```
Comment (sumeet-bansal) [inline package.json:23]: delete unused dependency

File: config.json.example [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (sumeet-bansal) [inline config.json.example:1]: delete unused config file since we're using env vars

File: modules/shorten.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
+/**
+ * @file shorten.js
+ * @description Module for shortening links using "acmurl" (Rebrandly API).
+ * @author Storm_FireFox1
+ */
+
+global.config = require('../config.json')
```
Comment (sumeet-bansal) [inline modules/shorten.js:7]: delete

File: server.js (hunk 5 above)
Comment (StormFireFox1) [reply server.js:25]: I thought we could keep it for later, but sure.

File: config.json.example (hunk 7 above)
Comment (StormFireFox1) [reply config.json.example:1]: I thought we could keep both options, but I'll switch everything up to envvars.

File: modules/shorten.js (hunk 8 above)
Comment (StormFireFox1) [reply modules/shorten.js]: All configuration is now done by environment variables, check 4219eda.

File: package.json (hunk 6 above)
Comment (StormFireFox1) [reply package.json]: Deleted, see f8c383f.

File: server.js (hunk 5 above)
Comment (StormFireFox1) [reply server.js]: Deleted, see f8c383f.

**Pull request merged**